sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ProactiveContainerInitConfigBuilder.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.List;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * A builder to build {@link ProactiveContainerInitConfig}
+ * */
+public final class ProactiveContainerInitConfigBuilder {
+
+    private static final int MAX_NO_OF_PROACTIVE_CONNECTION_REGIONS = 2;
+    private final List<CosmosContainerIdentity> cosmosContainerIdentities;
+    private int numProactiveConnectionRegions;
+
+    /**
+     * Instantiates {@link ProactiveContainerInitConfigBuilder}
+     *
+     * @param cosmosContainerIdentities the container identities - this parameter must be non-empty
+     */
+    public ProactiveContainerInitConfigBuilder(List<CosmosContainerIdentity> cosmosContainerIdentities) {
+        checkArgument(!cosmosContainerIdentities.isEmpty(), "The list of container identities should be non-empty.");
```

Comment:
Why disallow empty but not null?

Reply:
It's best to disallow both empty and null.